velox/functions/prestosql/JsonFunctions.cpp

```diff
@@ -63,10 +62,61 @@ class JsonFormatFunction : public exec::VectorFunction {
   }
 };
 
+class JsonParseFunction : public exec::VectorFunction {
+ public:
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    VectorPtr localResult;
+
+    // Input can be constant or flat.
+    assert(args.size() > 0);
+    const auto& arg = args[0];
+    if (arg->isConstantEncoding()) {
+      auto value = arg->as<ConstantVector<StringView>>()->valueAt(0);
+      folly::parseJson(value);
```

Comment:
Make sure to catch exception and report it using context.setErrors.

See https://facebookincubator.github.io/velox/develop/scalar-functions.html#try-expression-support